dashboard/client/dashboard.js

```diff
@@ -99,19 +112,53 @@ Template.dashboard.events({
       inputVars[key] = value;
     });
     Session.set('variableInputs', inputVars);
+  },
+  'click #addHeaderButton': function (event, template) {
+    var headerName = template.find('#newHeaderName').value;
+    var headerValue = template.find('#newHeaderValue').value;
+    headers =  ApiConfigs.findOne(activeConfigId.get()).headers;
+
+    if(_.contains(headers, headerName)) {
+      alert("Duplicate header"); //TODO: Replace with a reactive error
+    }
+    else {
+      template.find('#newHeaderName').value = "";
+      template.find('#newHeaderValue').value = "";
+
+      ApiConfigs.update(activeConfigId.get(), {$push: { headers: headerName }});
+      template.headerValues.set(headerName, headerValue);
+    }
+  },
+  'click #removeHeaderButton': function (event, template) {
+    ApiConfigs.update(activeConfigId.get(), {$pull: {headers: event.target.getAttribute('data-label')}} );
+  },
+  'keyup .headersName': function (event) {
+    // TODO update header in DB and in headerValus:
```

Comment:
OK I can't do that without merging this. But add `data-index='{{@index}}'` on the field in the template, and then here's the event handler:

```
  'keyup .headersName': function (event) {
    var index = event.target.dataset.index;
    var setter = {};
    setter[`headers.${index}`] = event.target.value;
    ApiConfigs.update(activeConfigId.get(), {$set: setter});
  },
```